webpack/ForemanTasks/Components/TasksTable/TasksTable.js

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect } from 'react';
+import { Spinner } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import { Table } from 'foremanReact/components/common/table';
+import { STATUS } from 'foremanReact/constants';
+import MessageBox from 'foremanReact/components/common/MessageBox';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { getURIQuery } from 'foremanReact/common/helpers';
+import Pagination from 'foremanReact/components/Pagination/PaginationWrapper';
+import createTasksTableSchema from './TasksTableSchema';
+import { updateURlQuery } from './TasksTableHelpers';
+import './TasksTable.scss';
+
+const TasksTable = ({
+  getTableItems,
+  error,
+  status,
+  results,
+  loading,
+  query,
+  itemCount,
+  pagination,
+}) => {
+  useEffect(() => {
+    getTableItems();
+  }, [query.state, query.result, query.mode, query.time]);
+  const onPageChange = paginationArgs => {
+    updateURlQuery({
+      page: paginationArgs.page,
+      per_page: paginationArgs.perPage,
+    });
+    getTableItems();
```

Comment:
Instead of calling the `getTableItems` action, wouldn't it make more sense to just update the state with the new `page` and `perPage` so then your `useEffect` will do it's job?

Reply:
Added setPagination action